### PR TITLE
Run specific set of tests in functional tests suite

### PR DESCRIPTION
**Pull request by eapolinario**
## Describe your changes
We pluck a specific set of tests and register those as opposed to registering all flytesnacks examples. 

This PR uses the default flytekit image as base image and fast-registers the flyte entities mentioned in the P0 functional tests.

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

File: .github/workflows/single-binary.yml [hunk 1]
```diff
@@ -175,19 +175,38 @@ jobs:
...
+                  control_flow/control_flow/subworkflows.py \
+                  type_system/type_system/custom_objects.py \
+                  type_system/type_system/schema.py \
+                  type_system/type_system/typed_schema.py ;
+          do
+              pyflyte --config ./boilerplate/flyte/end2end/functional-test-config.yaml \
+                  register \
+                  --project flytesnacks \
+                  --domain development \
+                  --image cr.flyte.org/flyteorg/flytekit:py3.9-latest \
```
Comment (jeevb) [inline .github/workflows/single-binary.yml:206]: Use latest version of `flytekit` image instead?

File: .github/workflows/single-binary.yml [hunk 2]
```diff
@@ -175,19 +175,38 @@ jobs:
...
       - name: Install Python dependencies
         run: |
           python -m pip install --upgrade pip
           pip install flytekit
           pip freeze
+      - name: Checkout flytesnacks
+        uses: actions/checkout@v3
+        with:
+          repository: flyteorg/flytesnacks
+          path: flytesnacks
```
Comment (jeevb) [inline .github/workflows/single-binary.yml:187]: Should we `FLYTESNACKS_VERSION` in this checkout step? We can update the variable with the latest commit sha if the original value was a branch like `master`.

Comment (eapolinario) [reply .github/workflows/single-binary.yml:187]: yup, left a comment in the next update explaining this.

File: .github/workflows/single-binary.yml (hunk 1 above)
Comment (eapolinario) [reply .github/workflows/single-binary.yml:208]: which image?

Comment (eapolinario) [reply .github/workflows/single-binary.yml:208]: sync'd offline. Meant latest python version image.

**Pull request merged**